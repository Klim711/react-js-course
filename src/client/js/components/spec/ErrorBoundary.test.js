import React from 'react';
import renderer from 'react-test-renderer';

import ErrorBoundary from '../ErrorBoundary';

describe('ErrorBoundary component', () => {
  let renderedValue;

  it('renders children if no errors', () => {
    renderedValue = renderer.create(
      <ErrorBoundary>
        mockCildren
      </ErrorBoundary>,
    ).toJSON();

    expect(renderedValue).toMatchSnapshot();
  });

  it('renders error message if error is happen', () => {
    const ErrorComponent = () => {
      throw new Error();
    };
    renderedValue = renderer.create(
      <ErrorBoundary>
        <ErrorComponent/>
      </ErrorBoundary>,
    ).toJSON();

    expect(renderedValue).toMatchSnapshot();
  });
});
