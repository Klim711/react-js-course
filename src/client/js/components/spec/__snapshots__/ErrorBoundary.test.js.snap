// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders children if no errors 1`] = `"mockCildren"`;

exports[`ErrorBoundary component renders error message if error is happen 1`] = `
<h1>
  Something went wrong.
</h1>
`;
